src/oracle/connection/mod.rs

```diff
@@ -130,44 +130,11 @@ impl Connection for OciConnection {
         U: Queryable<T::SqlType, Self::Backend>,
     {
         let mut stmt = self.prepare_query(&source.as_query())?;
-        if stmt.is_returning {
-            // the following are just taken from gst-database/insertable.rs and might help to determine the types of columns, idk yet
-            //type SqlType<T> = <T as ::diesel::expression::Expression>::SqlType;
-            //type AllColumns<T> = <T as ::diesel::query_source::Table>::AllColumns;
-
-            // the following is for now the easiest way to get the table name, if at some point @gese has better way that'd be great
-            let table = stmt.affected_table.clone();
-            // first we need to get the rowid, maybe the types T::SqlType can be replace with something else, like Text and U=String
-            let mut cursor: Cursor<::diesel::sql_types::Text, String> = stmt.run_with_cursor(self.auto_commit())?;
-
-            // let's read the rowid from there
-            let mut ret = Vec::new();
-            let rowid = cursor.next();
-            let rowid = rowid.unwrap()?;
-
-            // once we got it, use the row id to determine the proper row which we want to return
-            let sql = format!("select * from {} where rowid='{}'", table, rowid);
-            let query = ::diesel::sql_query(sql);
-            // this same as above
-            let mut stmt = self.prepare_query(&query)?;
-            let cursor2: Cursor<T::SqlType, U> = stmt.run_with_cursor(self.auto_commit())?;
-            // TODO: may we could use sth like this: `let stmt = self.prepare_query(AllColumns<source>.as_query());`
-
-            // this just reads the new cursor into ret
-            for el2 in cursor2 {
-                ret.push(el2?);
-            }
-
-
-            Ok(ret)
-        } else {
-            let cursor: Cursor<T::SqlType, U> = stmt.run_with_cursor(self.auto_commit())?;
-            let mut ret = Vec::new();
-            for el in cursor {
-                ret.push(el?);
-            }
-            Ok(ret)
-        }
+        let mut metadata = Vec::new();
+        // TODO: find a solution without deprecated function
```

Comment:
just for me, what's deprecated here?

Reply:
The function below, it is quite easy to remove :wink: